sympy/ntheory/ec.py

```diff
@@ -37,22 +37,25 @@ def __init__(self, a4, a6, a1=0, a2=0, a3=0, modulus = 0):
             domain = QQ
         else:
             domain = FF(modulus)
+        a1, a2, a3, a4, a6 = map(domain.convert, (a1, a2, a3, a4, a6))
         self._domain = domain
         domain = domain.convert
         self.modulus = modulus
         # Calculate discriminant
-        self._b2 = a1**2 + 4 * a2
-        self._b4 = 2 * a4 + a1 * a3
-        self._b6 = a3**2 + 4 * a6
-        self._b8 = a1**2 * a6 + 4 * a2 * a6 - a1 * a3 * a4 + a2 * a3**2 - a4**2
-        self._discrim = domain(-self._b2**2 * self._b8 - 8 * self._b4**3 - 27 * self._b6**2 + 9 * self._b2 * self._b4 * self._b6)
-        self._a1 = domain(a1)
-        self._a2 = domain(a2)
-        self._a3 = domain(a3)
-        self._a4 = domain(a4)
-        self._a6 = domain(a6)
-        self.x, self.y = symbols('x y')
-        self._eq = Eq(self.y**2 + self._a1*self.x*self.y + self._a3*self.y, self.x**3 + self._a2*self.x**2 + self._a4*self.x + self._a6)
+        b2 = a1**2 + 4 * a2
+        b4 = 2 * a4 + a1 * a3
+        b6 = a3**2 + 4 * a6
+        b8 = a1**2 * a6 + 4 * a2 * a6 - a1 * a3 * a4 + a2 * a3**2 - a4**2
+        self._b2, self._b4, self._b6, self._b8 = b2, b4, b6, b8
+        self._discrim = domain(-b2**2 * b8 - 8 * b4**3 - 27 * b6**2 + 9 * b2 * b4 * b6)
+        self._a1 = a1
+        self._a2 = a2
+        self._a3 = a3
+        self._a4 = a4
+        self._a6 = a6
+        x, y, z = symbols('x y z')
+        self.x, self.y, self.z = x, y, z
+        self._eq = Eq(y**2*z + a1*x*y*z + a3*y*z, x**3*z + a2*x**2*z + a4*x*z + a6*z)
```

Comment:
z is added to make this homogeneous of degree 3. So `a3*y*z**2`, `x**3 + a2*x**2*z + a4*x*z**2 + a6*z**3`.